src/endpoints/workers/create_route.rs

```diff
@@ -0,0 +1,29 @@
+use super::WorkersRouteIdOnly;
+
+use crate::framework::endpoint::{Endpoint, Method};
+
+/// Create a Route
+/// Creates a route mapping the given pattern to the given script
+/// https://api.cloudflare.com/#worker-routes-create-route
+pub struct CreateRoute<'a> {
+    pub zone_identifier: &'a str,
+    pub params: CreateRouteParams,
+}
+
+impl<'a> Endpoint<WorkersRouteIdOnly, (), CreateRouteParams> for CreateRoute<'a> {
+    fn method(&self) -> Method {
+        Method::Post
+    }
+    fn path(&self) -> String {
+        format!("zones/{}/workers/routes", self.zone_identifier)
+    }
+    fn body(&self) -> Option<CreateRouteParams> {
+        Some(self.params.clone())
+    }
+}
+
+#[derive(Serialize, Clone, Debug)]
+pub struct CreateRouteParams {
```

Comment:
This could perhaps just be called `Params` so people can do `create_route::Params` instead of `create_route::CreateRouteParams`.

Reply:
i'm into it; but following the pattern established by existing endpoints

Reply:
it may be worth making a separate issue for addressing repo-wide stutter in struct names.

Reply:
tbf, i actually export all public structs at the top level of the module anyway, so i import `workers::{CreateRoute, CreateRouteParams}` in wrangler for example